uarch/uarch-machine-state-access.h

```diff
@@ -411,10 +412,12 @@ class uarch_machine_state_access : public i_state_access<uarch_machine_state_acc
         raw_write_memory(shadow_state_get_csr_abs_addr(shadow_state_csr::iflags), val);
     }
 
-    void do_set_iflags_H(void) {
+    NO_RETURN NO_INLINE void do_set_iflags_H(void) {
         auto old_iflags = read_iflags();
         auto new_iflags = old_iflags | IFLAGS_H_MASK;
         write_iflags(new_iflags);
+        // The micro interpreter will never execute this line because it will halt as soon as iflags.H is set
+        __builtin_unreachable();
```

Comment:
But what happens due to a mistake of us (or the compiler), the code reaches here? I think we should make safe for that situation. `__builtin_unreachable()` will let the compiler do crazy optimizations here. I think we could use `__builtin_trap()` which essentially generates an illegal instruction, and this will cause our interpreter to abort.

In case of RISC-V 64, looks like `__builtin_trap()` generates `ebreak`  instruction:
https://gcc.godbolt.org/z/x3aMn4sq9

If doing this I think we should do the same for all other `__builtin_unreachable()`, that is, in `abort()`  and `interpret_next_mcycle_with_uarch()`.

Reply:
Depends what you mean by mistake. I don't know why we would "help" the compiler here by adding `_builtin_unreachable()`. The function is marked as [[noreturn]]. It writes to some "random" memory address through a volatile pointer. The compiler can't know what side effect that has. It really is a checkpoint, right? So it already knows everything it needs to know. I my mind, this is the same as a call to `exit` or something.

Reply:
>  I don't know why we would "help" the compiler here by adding _builtin_unreachable()

True, `__builtin_unreachable()` feels redundant in a function already marked as `[[noreturn]]`.

Reply:
__builtin_unreachable prevents the compiler from complaining about [[noreturn]] returning.
The exsisting test coverage catches  reordering of write_iflags() or write_mcycle() because it would change the expected mcycle.

Reply:
Ok, I am fine with `__builtin_unreachable()` to silence compiler warnings and given the fact that we have tests that will catch compiler write re-ordering.